<launch>
    <arg name="show" default="false" />
    <arg name="output" default="screen" />
    <arg name="send_img"     default="false"/>
    <arg name="send_whole_img_desc"     default="false"/>
    <arg name="is_4dof"     default="true"/>
    <arg name="self_id"     default="1"/>
    <!-- <arg name="superpoint_model_path"     default="$(find d2frontend)/models/superpoint_v1_rtx3080_512x512_fp16.trt"/> -->
    <!-- <arg name="netvlad_model_path"     default="$(find d2frontend)/models/mobilenetvlad_240x320_rtx3080_fp16.trt" /> -->
    <arg name="superpoint_model_path"     default="$(find d2frontend)/models/superpoint_v1_512x512.onnx"/>
    <arg name="netvlad_model_path"     default="$(find d2frontend)/models/mobilenetvlad_480x640.onnx" />
    <arg name="config" default="$(find d2vins)/../config/tum/tum_single.yaml" />
    <arg name="max_freq" default="1.0" />
    <arg name="match_index_dist" default="50" />
    <arg name="output_path" default="/home/dji/output/" />
    <arg name="min_loop_feature_num" default="20" />
    <arg name="init_loop_min_feature_num" default="20" />
    <arg name="nonkeyframe_waitsec" default="5.0" />
    <arg name="inter_drone_init_frames" default="3" />
    <arg name="enable_loop" default="false" />
    <arg name="rviz" default="false" />
    
    <node pkg="d2vins" type="d2vins_node" name="d2vins" output="screen" >
        <rosparam>
            lcm_uri: udpm://224.0.0.251:7667?ttl=1
            jpg_quality: 75
            
            query_thres: 0.2
            init_query_thres: 0.2

            min_movement_keyframe: 0.1
            enable_pub_remote_img: false
            enable_sub_remote_img: false
            accept_min_3d_pts: 30
            recv_msg_duration: 0.5
            superpoint_thres: 0.1
            triangle_thres: 0.012
            #triangle_thres: 0.008
            min_direction_loop: 1
            
            detector_match_thres: 0.7
            send_all_features: false
            lower_cam_as_main: false
            min_match_per_dir: 20
            output_raw_superpoint_desc: false
            depth_far_thres: 100
            loop_cov_pos: 0.1
            odometry_consistency_threshold: 3.0
            pos_covariance_per_meter: 0.01
            yaw_covariance_per_meter: 0.003
            enable_network: true
            ransacReprojThreshold: 10.0
            verbose: true
        </rosparam>
        <param name="pca_comp_path" value="$(find d2frontend)/models/components_.csv" type="string" />
        <param name="output_path" value="$(arg output_path)" type="string" />
        <param name="pca_mean_path" value="$(find d2frontend)/models/mean_.csv" type="string" />
        <param name="superpoint_model_path" value="$(arg superpoint_model_path)" type="string" />
        <param name="netvlad_model_path" value="$(arg netvlad_model_path)" type="string" />
        <param name="show" value="$(arg show)" type="bool" />
        <param name="send_img" value="$(arg send_img)" type="bool" />
        <param name="send_whole_img_desc" value="$(arg send_whole_img_desc)" type="bool" />
        <param name="self_id" value="$(arg self_id)" type="int" />
        <param name="match_index_dist" value="$(arg match_index_dist)" type="int" />
        <param name="nonkeyframe_waitsec" value="$(arg nonkeyframe_waitsec)" type="double" />
        <param name="min_loop_feature_num" value="$(arg min_loop_feature_num)" type="int" />
        <param name="init_loop_min_feature_num" value="$(arg init_loop_min_feature_num)" type="int" />
        <param name="max_freq" value="$(arg max_freq)" type="double" />
        <param name="inter_drone_init_frames" value="$(arg inter_drone_init_frames)" type="int" />
        <param name="vins_config_path" value="$(arg config)" type="string" />
        <param name="is_4dof" value="$(arg is_4dof)" type="bool" />
        <param name="enable_loop" value="$(arg enable_loop)" type="bool" />
        <remap from="~imu" to="/dji_sdk_1/dji_sdk/imu" />
    </node>
    <node name="rvizvisualisation" pkg="rviz" type="rviz" output="log" args="-d $(find d2vins)/../config/d2slam.rviz" if="$(arg rviz)" />
</launch>
